src/ApiProblem.php

```diff
@@ -215,20 +215,28 @@ protected static function decompile(array $parsed) : self
     {
         $problem = new static();
 
-
-        if (!empty($parsed['title'])) {
+        // Skip empty string or missing values.
+        // The string or integer 0, however, are allowed.  PHP makes
+        // this ugly. The check on string handles XML decompile which
+        // may return an empty array.
+        $emptyValues = ['', []];
+        if (isset($parsed['title']) && !in_array($parsed['title'], $emptyValues, true)) {
             $problem->setTitle($parsed['title']);
         }
-        if (!empty($parsed['type'])) {
+
+        if (isset($parsed['type']) && !in_array($parsed['type'], $emptyValues, true)) {
             $problem->setType($parsed['type']);
         }
-        if (!empty($parsed['status'])) {
-            $problem->setStatus((int) $parsed['status']);
+
+        if (isset($parsed['status']) && ($status = filter_var($parsed['status'], FILTER_VALIDATE_INT)) !== false) {
```

Comment:
Wouldn't `is_numeric()` and `intval()` work better here?  I suppose that would end up truncating a float, but if you send a float in here I wouldn't expect anything logical to happen.

(At some point I should bump this library to PHP 7.4 so I can type the properties.  Not right now, though.)

Reply:
I will use my alternate solution with `intval` and `strval` which will also reject `float`.